libcaf_net/caf/net/consumer_queue.hpp

```diff
@@ -34,7 +34,7 @@
 
 namespace caf::net {
 
-class CAF_NET_EXPORT endpoint_manager_queue {
+class CAF_NET_EXPORT consumer_queue {
```

Comment:
The name suggests to me that this class is very generic queue type for (any number of?) producers to write to that belongs to a single consumer. But that just isn't the case. The implementation is in fact very specifically tailored to the needs of the BASP application. I think it shouldn't be in the public `caf::net` API to begin with and the name gives no guidance as to what role this class actually addresses.